### PR TITLE
[fix issue #724] replace panic! with return Err()

### PR DESCRIPTION
**Pull request by pventuzelo**
related to issue: https://github.com/wasmerio/wasmer/issues/724

after the patch:
```
./target/release/wasmer run reported_issues/panic_wasm_trap_occured_call_indirect.wasm 
sizeof(UAFME) = 4
execute_wasm: "wasm trap occured: `call_indirect` out-of-bounds"
```

### CODE REVIEW

File: src/bin/wasmer.rs [hunk 1]
```diff
@@ -644,7 +644,7 @@ fn execute_wasm(options: &Run) -> Result<(), String> {
                         #[cfg(not(feature = "wasi"))]
                         RuntimeError::Error { .. } => (),
                     }
-                    panic!("error: {:?}", err)
+                    return Err(format!("error: {:?}", err))
```
Comment (syrusakbary) [inline src/bin/wasmer.rs]:
```suggestion
return Err(format!("error: {:?}", err));
```

File: src/bin/wasmer.rs [hunk 2]
```diff
@@ -634,7 +634,7 @@ fn execute_wasm(options: &Run) -> Result<(), String> {
 
                 if let Err(ref err) = result {
                     match err {
-                        RuntimeError::Trap { msg } => panic!("wasm trap occured: {}", msg),
+                        RuntimeError::Trap { msg } => return Err(format!("wasm trap occured: {}", msg)),
```
Comment (syrusakbary) [inline src/bin/wasmer.rs]:
```suggestion
                        RuntimeError::Trap { msg } => return Err(format!("wasm trap occured: {}", msg)),
                        RuntimeError::Trap { msg } => {
                            return Err(format!("wasm trap occured: {}", msg))
                        }
```

**Comment by syrusakbary** [comment]
@pventuzelo there are lint errors on this PR, once the suggestions are merged it should be good to go.

For next times run `cargo fmt --all` and cargo will autoformat it for you

**Comment by pventuzelo** [comment]
@syrusakbary sorry for the lint error, completely forgot to `fmt` before pull request, thx.

**Pull request merged**